forest/bonsai/simulate_gps_data.py

```diff
@@ -1117,31 +1122,261 @@ def process_switches(
     return switches
 
 
-def sim_GPS_data(cycle,p,data_folder):
-    ## only two parameters
-    ## cycle is the sum of on-cycle and off_cycle, unit is minute
-    ## p is the missing rate, in other words, the proportion of off_cycle, should be within [0,1]
-    ## it returns a pandas dataframe with only observations from on_cycles, which mimics the real data file
-    ## the data are the trajectories over two weeks, sampled at 1Hz, with an obvious activity pattern
-    s = datetime2stamp([2020,8,24,0,0,0],'America/New_York')*1000
-    if os.path.exists(data_folder)==False:
+def sim_gps_data(
+    N: int,
+    location: str,
+    start_date: List[int],
+    end_date: List[int],
+    cycle: int,
+    percentage: float,
+    api_key: str,
+    data_folder: str,
+    attributes_dir: Optional[str] = None,
+):
+    """Generates gps trajectories.
+
+    Args:
+        N: (int) number of people to simulate
+        location: (str) indicating country and city to simulate at,
+            format "Country_2_letter_ISO_code/City_Name"
+        start_date: (list) start date of trajectories,
```

Comment:
This is an unusual way to deal with a date. Would it make more sense to make it a datetime date?
https://docs.python.org/3.6/library/datetime.html#datetime.date